src/store/actions/contactsActions.js

```diff
@@ -1,87 +1,132 @@
-
 export const getContactsByCurrentUser = () => {
   return (dispatch, getState, { getFirestore }) => {
-    const firestore = getFirestore()
-    const userID = getState().firebase.auth.uid // later for mapping userid to contacts
+    const firestore = getFirestore();
+    const userID = getState().firebase.auth.uid; // later for mapping userid to contacts
     // onSnapshot - try to implement for auto-updating
-    firestore.collection('users').where('uid', '==', userID).get().then(querySnapshot => {
-      let userInfo = []
-      querySnapshot.forEach(doc => {
-        userInfo.push(doc.data())
+    firestore
+      .collection("users")
+      .where("uid", "==", userID)
+      .get()
+      .then(querySnapshot => {
+        let userInfo = [];
+        querySnapshot.forEach(doc => {
+          userInfo.push(doc.data());
+        });
+        const contactArray = userInfo[0].contacts;
+        dispatch({ type: "GET_CONTACTS", contactArray });
+      })
+      .catch(err => {
+        dispatch({ type: "GET_CONTACTS_ERROR", err });
       });
-      const contactArray = userInfo[0].contacts;
-      dispatch({ type: "GET_CONTACTS", contactArray }
-      );
-    })
-    .catch((err) => {
-      dispatch({ type: "GET_CONTACTS_ERROR", err})
-    })
-
   };
 };
 
 export const addToContacts = (searchedEmail, currentUserUid) => {
   return (dispatch, getState, { getFirestore }) => {
-    const firestore = getFirestore()
-    const userID = getState().firebase.auth.uid // later for mapping userid to contacts
+    const firestore = getFirestore();
+    const userID = getState().firebase.auth.uid; // later for mapping userid to contacts
     // onSnapshot - try to implement for auto-updating
-    console.log(userID)
-    firestore.collection('users').where('uid', '==', userID).get().then(querySnapshot => {
-      let userInfo = []
-      querySnapshot.forEach(doc => {
-        console.log(doc.data())
-        userInfo.push(doc.data())
+    Promise.all([
+      firestore
+        .collection("users")
+        .where("uid", "==", userID)
+        .get(),
+      firestore
+        .collection("users")
+        .where("uid", "==", searchedEmail[0].uid)
+        .get()
+    ]).then(profiles => {
+      const profilesArray = [];
+      profiles.forEach(profile => {
+        profile.forEach(querySnapshot => {
+          const data = querySnapshot.data();
+          profilesArray.push(data);
+        });
+      });
+      const hasContact = profilesArray[1].contacts.filter(contact => {
+        return contact.uid === profilesArray[0].uid;
+      });
+      const hasContactTwo = profilesArray[0].contacts.filter(contact => {
+        return contact.uid === profilesArray[1].uid;
+      });
+
+      if (hasContact.length === 0) {
+        let removedContact = { ...profilesArray[0] };
+        delete removedContact["contacts"];
+        profilesArray[1].contacts.push(removedContact);
+      }
```

Comment:
"removedContact" stores the information of the person you're adding to the other persons contact.
I understand you then remove the contact data from it before you push it, but maybe a different name (like "newContactData") would be more logical?